core/string/ustring.cpp

```diff
@@ -3840,7 +3834,7 @@ String String::rstrip(const String &p_chars) const {
 		return *this;
 	}
 
-	return substr(0, end + 1);
+	return left(end + 1);
```

Comment:
`end` goes through the loop starting as `length() - 1` and the loop ends when it's less than 0. If the string is empty it might end up being -1, but this case returns early in a condition after the loop.